feat(common): :sparkles: create application, add log in, registration…

romanchukegor opened the pull request:
… and logout functions.

Create log in, registration and logout functions, make adaptive, done absolute imports, use store, make validation, connect to database

Commit:
feat(common): :sparkles: create application, add log in, registration and logout functions.

Create log in, registration and logout functions, make adaptive, done absolute imports, use store, make validation, connect to database

baklanovairina commented on src/App.js:
```diff
@@ -1,9 +1,36 @@
+import Authorization from "./components/Authorization/Authorization";
+import Home from "./components/Home/Home";
+import Registration from "./components/Registration/Registration";
+import "./interceptors/axios";
```
это зачем здесь?

baklanovairina commented on src/App.js:
```diff
@@ -1,9 +1,36 @@
+import Registration from "./components/Registration/Registration";
+import "./interceptors/axios";
 
 function App() {
```
стрелочная функция

baklanovairina commented on src/App.js:
```diff
@@ -1,9 +1,36 @@
 
 function App() {
+  const { store } = useContext(Context);
+  const [isAuthic, setIsAuthic] = useState(store.isAuth);
```
что тако isAuthic? Лучше переименуй

baklanovairina commented on src/App.js:
```diff
@@ -1,9 +1,36 @@
+
+  useEffect(() => {
+    store.checkAuth();
+    store.subscribe("isAuth", (params) => setIsAuthic(params));
```
уверен, что порядок вызовов функций верный?
params - тоже переименуй, мы здесь не просто параметры передаем какие-то

baklanovairina commented on src/App.js:
```diff
@@ -1,9 +1,36 @@
   return (
-    <div className="App">
-      
+    <div>
```
зачем див?

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+import { useContext, useState } from "react";
+import { Link } from "react-router-dom";
+import Error from "components/Error/Error";
+import { Context } from "../..";
```
порядок: контекст импортируется раньше компонентов
и путь поправь, не должно быть этого ../

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+import Error from "components/Error/Error";
+import { Context } from "../..";
+import HeaderImage from "components/HeaderImage/HeaderImage";
+import Build from "images/Build.svg";
```
названия картинок обычно с маленькой буквы

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+
+const Authorization = () => {
+  const [login, setLogin] = useState("");
+  const [password, setPassword] = useState("");
```
вместо двух стейтов сделай один в виде объекта

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+  const [errorMessage, setErrorMessage] = useState("");
+  const { store } = useContext(Context);
+
+  const userLogin = async () => {
```
а где глагол? Что делает функция - не понятно

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+  const { store } = useContext(Context);
+
+  const userLogin = async () => {
+    if (login === "" || password === "") {
```
во первых, не верная проверка на пустоту, пробелы не учитываешь
во- вторых, пользователь точно должен знать, какое поле не верно заполнено, а не гадать, где ошибка

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+      setIsError(true);
+      setErrorMessage("Все поля должны быть заполнены");
+    } else {
+      const response = await store.login(login, password);
```
ты либо ждешь ошибку, либо объект, что-то одно и работаешь с этим

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+      const response = await store.login(login, password);
+
+      if (!response.data) {
+        setErrorMessage("Ошибка входа");
```
а где setIsError? Откроется ли тогда снэкбар?

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+    <div className="user-form">
+      <div className="user-form__header">
+        <HeaderImage />
+        <div className="user-form__title">Войти в систему</div>
```
передавай это в Header в виде пропсов, у тебя же хэдер везде одинаковый, а что отличается - передавай в пропсах

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+        </div>
+        <div className="authorization">
+          <div className="authorization__title">Войти в систему</div>
+          <div>
```
почему это не форма?

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+        <div className="authorization">
+          <div className="authorization__title">Войти в систему</div>
+          <div>
+            <p>Логин:</p>
```
это label

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+                  : "authorization__input__error"
+              }
+              placeholder="Логин"
+              onChange={(e) => setLogin(e.target.value)}
```
напиши один handleChange для полей этого компонента и передавай в него имя поля строкой и e.target.value

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+              onChange={(e) => setLogin(e.target.value)}
+            />
+          </div>
+          <div>
```
в общем дальше всё аналогично

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,88 @@
+              onChange={(e) => setPassword(e.target.value)}
+            />
+          </div>
+          <div>
```
вот тут прям класс не нужен? А как ты задаешь размещение кнопки и ссылки ниже?

baklanovairina commented on src/components/Authorization/style.scss:
```diff
@@ -0,0 +1,94 @@
+  height: 450px;
+  background-color: white;
+  border-radius: 5px;
+  &__input {
```
делай отступы между классами везде

baklanovairina commented on src/components/Authorization/style.scss:
```diff
@@ -0,0 +1,94 @@
+  }
+}
+
+.user-form {
```
что-то не понятно, сначала был user-form, а потом вдруг authorization? А в чем разница тогда? По сути это одно и тоже

baklanovairina commented on src/components/Error/Error.jsx:
```diff
@@ -0,0 +1,14 @@
+import { Snackbar } from "@mui/material";
+
+const Error = ({ errorMessage, isError }) => {
+  const [open] = useState(isError);
```
ээ.. что? какой-то не дописанный useState и что open? что открывается. Да и вообще зачем он?

baklanovairina commented on src/components/Error/Error.jsx:
```diff
@@ -0,0 +1,14 @@
+
+  return (
+    <div>
+      <Snackbar open={open} autoHideDuration={3000} message={errorMessage} />
```
код стайл - перенос строк

baklanovairina commented on src/components/HeaderImage/HeaderImage.jsx:
```diff
@@ -0,0 +1,12 @@
+import Vector from "images/Vector.svg";
+import "./style.scss";
+
+const HeaderImage = () => {
```
так то хэдер включает и заголовок и кнопку на странице всех записей на прием, а не только лого

baklanovairina commented on src/components/HeaderImage/HeaderImage.jsx:
```diff
@@ -0,0 +1,12 @@
+import Vector from "images/Vector.svg";
```
картинки с маленькой буквы

baklanovairina commented on src/components/HeaderImage/style.scss:
```diff
@@ -0,0 +1,3 @@
+.hospital-image {
```
наименование класса не говорит к какому компоненту он принадлежит

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from "react";
```
React?

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from "react";
+import { useState } from "react";
```
зачем дважда? Нелья объединить строчки?

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,28 @@
+import React, { useContext } from "react";
+import { useState } from "react";
+import { Context } from "../..";
```
выше писала

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,28 @@
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const userLogout = async () => {
+    const response = await store.logout();
```
то же самое, что ранее писала, ждем либо ошибку либо респонс

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <div>
+      <button onClick={userLogout}>logout</button>
+      {isError && <Error errorMessage={errorMessage} isError={isError} />}
```
код стайл

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,102 @@
+import { useState, useContext } from "react";
```
в этом компоненте те же ошибки, что и в авторизации, дублировать не буду

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,102 @@
+  const { store } = useContext(Context);
+
+  const userRegistration = async () => {
+    if (login === "" || password === "" || repeatedPassword === "") {
```
валидацию вообще объедини, зачем ты сначала все поля проверяешь на одно, потом на другое. 
Ты должен проверять сначала поле логин на всё что нужно
Потом поле пароля, причем пароль будет регулярке, почитай тех задание внимательно, там не просто 6 символов должно быть

baklanovairina commented on src/components/Registration/style.scss:
```diff
@@ -0,0 +1,119 @@
+  }
+}
+
+.user-form {
```
зачем столько раз дублировать стили?

baklanovairina commented on src/index.js:
```diff
@@ -1,14 +1,24 @@
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import React, { createContext } from "react";
```
React?

baklanovairina commented on src/index.js:
```diff
@@ -1,14 +1,24 @@
+import "./index.css";
+import App from "./App";
+import reportWebVitals from "./reportWebVitals";
+import { BrowserRouter } from "react-router-dom";
```
порядок тоже нарушен

baklanovairina commented on src/index.js:
```diff
@@ -1,14 +1,24 @@
-    <App />
-  </React.StrictMode>
+  <BrowserRouter>
+    <Context.Provider value={{store}}>
```
лишние скобки есть

и почему BrowserRouter не берешь? Тебе не все компоненты нужны для контекста?

baklanovairina commented on src/index.js:
```diff
@@ -1,14 +1,24 @@
-const root = ReactDOM.createRoot(document.getElementById('root'));
+const store = new Store();
+
+export const Context = createContext({
```
лишние скобки есть, ты в итоге объект в объекте создаешь

baklanovairina commented on src/interceptors/axios.js:
```diff
@@ -0,0 +1,15 @@
+import axios from "axios";
+
+export const API_URL = "http://localhost:8000/api/";
```
это в константы

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,19 @@
+import api from "../interceptors/axios";
+
+export default class AuthService {
```
обоснуй, почему по дефолту?

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,19 @@
+import api from "../interceptors/axios";
+
+export default class AuthService {
+  static async login(login, password) {
```
и также обоснуй, почему статические методы используешь

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,19 @@
+  static async login(login, password) {
+    return api
+      .post("login", { login, password })
+      .then((response) => response.data);
```
лучше async/await используй, а не then

вообще странно, везде async написал, а внутри в итоге then используешь, как будто не определился

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+
+export default class Store {
+  isAuth = false;
+  apiErrorMessage = "";
```
а что это такое и зачем? Ошибки то ты в компоненты возвращаешь для обработки и вывода на экран

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+export default class Store {
+  isAuth = false;
+  apiErrorMessage = "";
+  isApiError = false;
```
аналогично, что и выше, непонятно для чего это

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+  isAuth = false;
+  apiErrorMessage = "";
+  isApiError = false;
+  user = {};
```
почему пустой? Может null? 
Просто так то объект всё равно не пустой

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+  apiErrorMessage = "";
+  isApiError = false;
+  user = {};
+  channels = {};
```
каналы чего? Для чего? и почему объект? По-моему это массив по паттерну

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+  user = {};
+  channels = {};
+
+  setAuth = (bool) => {
```
не сокращай boolean

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+    channel.forEach((listener) => listener(data));
+  }
+
+  login = async (login, password) => {
```
ты определись лучше как ты пишешь методы в классе так: 
`login = async (login, password) => { `
или так
`publish(channelName, data) {`
а то где-то через стрелочную, где-то нет

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+      localStorage.setItem("token", response.accessToken);
+      this.setAuth(true);
+      this.setUser(response.user);
+      return response;
```
а надо ли нам response возвращать или всё же достаточно ошибки? С ошибкой мы хотя бы работает и выводим на экран и на основе нее можем и проверить всё ли успешно прошло. Тогда зачем на response?

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+      this.setUser(response.user);
+      return response;
+    } catch (error) {
+      return error;
```
не возвращай целиком объект, достаточно текста ошибки, 
либо уже в той функции куда возвращаешь, вытаскивай тест ошибки иначе в снэкбар полетит куча лишней информации, не нужно пользователю

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+    }
+  };
+
+  registration = async (login, password) => {
```
в ниже функциях те же ошибки, что и выше

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,92 @@
+        this.setAuth(true);
+        this.setUser(response.user);
+        return response;
+      }
```
а что делать если в ветку с if не зайдет, но и в catch не попадет? Может стоит почистить юзера и Auth?

Commit:
fix(common): :bug: Fix code style and bugs

Fix validation, correct code style, rename some elements, create helpers and constants, fix error handler, correct snackbar

baklanovairina commented on src/App.js:
```diff
@@ -1,10 +1,34 @@
+  const [isAuthentication, setIsAuthentication] = useState(store.isAuth);
+
+  useEffect(() => {
+    store.subscribe("isAuth", (boolean) => setIsAuthentication(boolean));
```
назови по другому параметр, не boolean

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+import { useContext, useState } from "react";
+import { Link } from "react-router-dom";
+import { Context } from "index";
```
везде неявно указываем путь в Context

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+import { Context } from "index";
+import Error from "components/Error/Error";
+import HeaderImage from "components/Header/Header";
+import Helpers from "helpers/helpers";
```
тебе не кажется, что как то многовато Helpers используется? 
имя класса точно поменяй, вообще не понятно для чего он и что в нем хранится
потом, что это helpers/helpers? Почему папка и файл одинаково называются? Что тоже в этом файле лежит непонятно

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+import Error from "components/Error/Error";
+import HeaderImage from "components/Header/Header";
+import Helpers from "helpers/helpers";
+import build from "images/Build.svg";
```
/Build.svg
ну так то файл тоже с маленькой буквы называется

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+import build from "images/Build.svg";
+import "./style.scss";
+
+const Authorization = () => {
```
как я уже кому то говорила, здесь юзается логин, это не авторизация, или хотя бы форма логина

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+  const [errorMessage, setErrorMessage] = useState("");
+  const store = useContext(Context);
+
+  const userSignIn = async () => {
```
то SignIn, то Authorization, то Login
что-то одно выбери

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+    if (Helpers.checkLogin(user.login)) {
+      setIsError(true);
+      setErrorMessage("Поле для логина должно быть больше 6 символов");
+    } else if (!Helpers.checkPassword(user.password)) {
```
ну нет, избавляйся от этих else if/else 
они излишне и ухудшают читаемость

baklanovairina replied:
что то у тебя с функциями не так из хэлперс
почему одна возвращает true и это ошибка, а другая false при ошибки
конечно вторая корректнее построена, так как логично, что  false это ошибка

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+      setErrorMessage("Поле для логина должно быть больше 6 символов");
+    } else if (!Helpers.checkPassword(user.password)) {
+      setIsError(true);
+      setErrorMessage("Пароль веден не корректно");
```
эти 2 строки повторяются часто

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+    } else {
+      const error = await store.login(user.login, user.password);
+
+      if (error) {
```
еще и вложенный if? 
тоже подумай как избавится, мы уже это проходили

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+    }
+  };
+
+  const handleChange = (event) => {
```
почему event? передавай сразу сюда и name и event.target.value

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+  };
+
+  return (
+    <div className="authentication-page">
```
ну не authentication это вовсе, у тебя компонент авторизации, а не аутентификации

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+      </div>
+      <div className="authentication-page__body">
+        <div>
+          <img src={build} alt="" className="authentication-page__image" />
```
код стайл

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+        <div className="authorization-form">
+          <div className="authorization-form__title">Войти в систему</div>
+          <form className="authorization-form__login">
+            <label>Логин:</label>
```
htmlFor добавь

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+            />
+          </form>
+          <form className="authorization-form__password">
+            <label>Пароль:</label>
```
htmlFor

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+              onChange={handleChange}
+            />
+          </form>
+          <form className="authorization-form__password">
```
зачем второй раз form?
Так, что такое форма?

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+              className={
+                !isError
+                  ? "authorization-form__input"
+                  : "authorization-form__input__error"
```
что-то тут с неймингом последнего класса не так, не совсем по БЭМ

baklanovairina commented on src/components/Authorization/Authorization.jsx:
```diff
@@ -0,0 +1,105 @@
+              Войти
+            </button>
+            <Link to="/registration">
+              <button className="registration-link" type="button">
```
нейминг класса, это не блок уже, а элемент

baklanovairina commented on src/components/Error/Error.jsx:
```diff
@@ -0,0 +1,14 @@
+
+const Error = ({ errorMessage, isError }) => {
+  return (
+    <div>
```
зачем тут див?

baklanovairina commented on src/components/Error/Error.jsx:
```diff
@@ -0,0 +1,14 @@
+  return (
+    <div>
+      <Snackbar 
+      open={isError} 
```
табы

baklanovairina commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,13 @@
+import vector from "images/Vector.svg";
```
Vector.svg выше писала

baklanovairina commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,13 @@
+import vector from "images/Vector.svg";
+import "./style.scss";
+
+const HeaderImage = ({ title }) => {
```
во-первых, не может отличаться название компонента от названия файла
во-вторых, это не HeaderImage

baklanovairina commented on src/components/Header/style.scss:
```diff
@@ -0,0 +1,22 @@
+.header {
+  display: flex;
+  margin-left: 50px;
```
отступ

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,30 @@
+  return (
+    <div>
+      <button onClick={userLogout}>logout</button>
+      {isError && <Error 
```
код стайл поправь

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,126 @@
+import { Context } from "index";
+import HeaderImage from "components/Header/Header";
+import build from "images/Build.svg";
+import Error from "components/Error/Error";
+import Helpers from "helpers/helpers";
```
порядок нарушен

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,126 @@
+  const [errorMessage, setErrorMessage] = useState("");
+  const store = useContext(Context);
+
+  const userRegistration = async () => {
```
если это функция зарегистрировать пользователя, то порядок слов не верный, сначала глагол идет

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,126 @@
+  const store = useContext(Context);
+
+  const userRegistration = async () => {
+    if (Helpers.checkLogin(user.login)) {
```
ну в целом здесь те же проблемы, что и в другом компоненте

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,126 @@
+          <div className="registration-form__title">
+            Зарегистрироваться в системе
+          </div>
+          <div>
```
а почему здесь уже див, а не форма?

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,126 @@
+  };
+
+  return (
+    <div className="authentication-page">
```
все тоже самое, что и выше писала

baklanovairina commented on src/helpers/helpers.js:
```diff
@@ -0,0 +1,17 @@
+  }
+}
+
+// we use static to store data at the class level.
```
почему комменты внизу? 
почитай на досуге
https://github.com/leonidlebedev/javascript-airbnb/tree/master/react

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,17 @@
+import api from "../interceptors/axios";
```
путь

baklanovairina replied:
а названия для папки и файла ты откуда взял?

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,85 @@
+        localStorage.removeItem("token", response.accessToken);
+        this.setAuth(true);
+        this.setUser(response.user);
+      }
```
прошлый коммент не исправил? Где изменение Auth если токена нет?

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,85 @@
+      this.setAuth(true);
+      this.setUser(response.user);
+    } catch (error) {
+      return error.message;
```
а ты уверен, что там объект всегда в таком виде приходит? Тестил?

Commit:
fix(common): :bug: Fix bugs, code style, and files structure

Fix bugs, correct code style, rename some files and folders, fix validators, improve code structure

baklanovairina commented on src/App.js:
```diff
@@ -1,10 +1,34 @@
+  const [isAuthentication, setIsAuthentication] = useState(store.isAuth);
+
+  useEffect(() => {
+    store.subscribe("isAuth", (checkAuth) => setIsAuthentication(checkAuth));
```
checkAuth - это вроде параметр + у тебя так ниже функция называется, поэтому лучше переназвать

baklanovairina commented on src/components/Error/Error.jsx:
```diff
@@ -0,0 +1,12 @@
+    <Snackbar 
+      open={isError} 
+      autoHideDuration={3000} 
+      message={errorMessage} />
```
```
  message={errorMessage} 
/>
```

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,29 @@
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const userLogout = async () => {
```
нейминг

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,29 @@
+
+  return (
+    <div>
+      <button onClick={userLogout}>logout</button>
```
а эта кнопка разве не часть хэдера?

baklanovairina commented on src/helpers/validator.js:
```diff
@@ -0,0 +1,19 @@
+// we use default to export only the class.
+export default class Validator {
```
почему класс называется здесь Validator, а везде ты его импортишь как Helpers

baklanovairina commented on src/helpers/validator.js:
```diff
@@ -0,0 +1,19 @@
+// we use default to export only the class.
```
это не объясняет почему ты решил использовать класс

baklanovairina commented on src/helpers/validator.js:
```diff
@@ -0,0 +1,19 @@
+  // we use static to store data at the class level.
+  static checkEqualPasswords(password, repeatedPassword) {
+    return repeatedPassword !== password;
+  }
```
лепишь код

baklanovairina commented on src/helpers/validator.js:
```diff
@@ -0,0 +1,19 @@
+// we use default to export only the class.
+export default class Validator {
+  // we use static to store data at the class level.
+  static checkEqualPasswords(password, repeatedPassword) {
```
почему привязываешь названиями эти функции-помощники только к двум компонентам. Так то я могу использовать эти функции где угодно, на то они и в хэлперс лежат

baklanovairina commented on src/helpers/validator.js:
```diff
@@ -0,0 +1,19 @@
+    return login.trim().length > 6;
+  }
+  static checkPassword(password) {
+    return /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{6,}$/g.test(
```
а если я захочу проверить строки на штук еще 10 регулярных выражение, мне тогда писать функцию для каждой регулярки? Как думаешь, это оптимально?

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,118 @@
+  return (
+    <div className="authorization-page">
+      <div className="authorization-page__header">
+        <HeaderImage title={"Войти в систему"} />
```
HeaderImage? откуда этот компонент, она разве не Header?

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,118 @@
+      </div>
+      <div className="authorization-page__body">
+        <div>
+          <img src={build} alt="" className="authorization-page__image" />
```
код стайл, говорила уже, что делать если атрибутов больше 2х

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,118 @@
+        </div>
+        <div className="authorization-form">
+          <div className="authorization-form__title">Войти в систему</div>
+          <form className="authorization-form__form">
```
какая то тавтология

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,118 @@
+              onClick={logInUser}
+              type="button"
+            >
+              Войти
```
кнопка вообще то должна входить в элементы формы, а не жить отдельной жизнью
ты ведь при нажатии на нее отправляешь данные из формы

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,118 @@
+          </div>
+        </div>
+      </div>
+      {isError && <Error errorMessage={errorMessage} isError={isError} />}
```
код стайл

baklanovairina commented on src/components/LogInForm/style.scss:
```diff
@@ -0,0 +1,124 @@
+.authorization-form {
```
по моему authorization-form это уже вложенный класс в authorization-page

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,134 @@
+import { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import { Context } from "src";
+import HeaderImage from "src/components/Header/Header";
```
опять HeaderImage

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,134 @@
+      handleError("Пароль введен не корректно");
+      return;
+    }
+    if (!Validator.checkEqualPasswords(user.password, user.confirmPassword)) {
```
про отступы снова забыл

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,134 @@
+    const error = await store.registration(user.login, user.password);
+
+    if (error) {
+      handleError(error);
```
а почему в логин ты ретернешь, а здесь нет? Где верно, а где нет?

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,134 @@
+    }
+  };
+
+  return (
```
повторяться не буду, здесь те же ошибки

baklanovairina commented on src/components/Registration/style.scss:
```diff
@@ -0,0 +1,142 @@
+  }
+}
+
+.registration-page {
```
выше писала уже о вложенности

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,17 @@
+
+export default class AuthService {
+  static async login(login, password) {
+    const response = await api.post("login", { login, password });
```
а куда слэш в пути подевался? В константе остался?

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,17 @@
+export default class AuthService {
+  static async login(login, password) {
+    const response = await api.post("login", { login, password });
+    return response.data;
```
это ты зря, рановато достаешь data
может не корректно отработать, если этого свойства не будет, в итоге может и ошибка не словиться дальше

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,89 @@
+      this.setAuth(true);
+      this.setUser(response.user);
+    } catch (error) {
+      console.log(error.response);
```
никаких консолей на фронте

Commit:
fix(common): :art: Fix code style, correct helpers, delete indents, rename some elements

Commit:
fix(common): :art: Fix code style, correct classnames

baklanovairina commented on src/components/Header/style.scss:
```diff
@@ -0,0 +1,31 @@
+    margin-left: 50px;
+    font-size: 36px;
+    font-weight: 300;
+  }
```
отступ забыл

baklanovairina commented on src/components/Header/style.scss:
```diff
@@ -0,0 +1,31 @@
+  }
+  &-box {
+    display: flex;
+    font-size: 24px;
```
отступ

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,32 @@
+  return (
+    <div>
+      <Header title={"Приемы"}>
+        <button onClick={logOutUser}>Выход</button>
```
тип кнопки

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,126 @@
+import { useContext, useState } from "react";
+import { Link } from "react-router-dom";
+import { Context } from "src";
+import { regexForPassword } from "src/constants";
```
константы после компонентов идут

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,126 @@
+            Войти в систему
+          </div>
+            <div className="login-body-form-inputs">
+            <label htmlFor="login">Логин:</label>
```
табы поправь

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,142 @@
+import { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import { Context } from "src";
+import { regexForPassword } from "src/constants";
```
порядок тоже

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,142 @@
+  });
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const store = useContext(Context);
```
отступы

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,142 @@
+              }
+            />
+            </div>
+            <div className="registration-body-form-buttons">
```
везде здесь проверь табы

Commit:
refactor: Fix code style and regex

baklanovairina commented on src/components/Registration/Registration.jsx:
```diff
@@ -0,0 +1,146 @@
+import Header from "src/components/Header/Header";
+import Error from "src/components/Error/Error";
+import { regexForPassword } from "src/constants";
+import {checkStringByRegex, checkStringLength, checkStringsEquals} from "src/helpers/validator";
```
здесь пробелы забыл

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,17 @@
+
+export default class AuthService {
+  static async login(login, password) {
+    const response = await api.post("/login", { login, password });
```
перед ретерн отступ добавь

baklanovairina commented on src/services/AuthService.js:
```diff
@@ -0,0 +1,17 @@
+  }
+
+  static async registration(login, password) {
+    const response = await api.post("/registration", { login, password });
```
и здесь перед ретерн отступ добавь

Commit:
refactor: add indents

baklanovairina commented on src/components/LogInForm/style.scss:
```diff
@@ -0,0 +1,120 @@
+      margin-bottom: 24px;
+    }
+
+      &-buttons__button {
```
а где класс login-body-form-buttons, в который должен быть вложен как раз этот класс? В компоненте этот класс login-body-form-buttons есть, а здесь нет

baklanovairina commented on src/components/LogInForm/style.scss:
```diff
@@ -0,0 +1,120 @@
+        font-weight: 300;
+      }
+
+      &-buttons__registration-link {
```
та же ситуация

Commit:
fix: fix code style

Commit:
fix: fix code style

Commit:
fix: Fix code style and rename registration component

Commit:
fix(common): :bug: correct code style, fix pubsub functions, add types and pointers for button

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,39 @@
+        <button 
+          onClick={logOutUser} 
+          type="button" 
+          className="home__button">
```
это home__button 
кнопка же находится внутри компонента Header 
и класс хоть и вызывается здесь, прописывается в стилях Header

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,121 @@
+import Header from "src/components/Header/Header";
+import { regexForPassword } from "src/constants";
+import { checkStringByRegex, checkStringLength } from "src/helpers/validator";
+import build from "src/images/build.svg";
```
build - странное название у картинки, переименуй

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,121 @@
+        <Header title={"Войти в систему"} />
+      </div>
+      <div className="login-body">
+        <div>
```
а зачем этот див? у тебя в нем только img один

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,121 @@
+          <img 
+            src={build} 
+            alt="" 
+            className="login-body__image" />
```
```
  className="login-body__image" 
/>
```

baklanovairina commented on src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,143 @@
+  return (
+    <div className="registration">
+      <div className="registration-header">
+        <Header title={"Зарегистрироваться в системе"} />
```
не верный синтаксис в пропсах

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,121 @@
+  return (
+    <div className="login">
+      <div className="login-header">
+        <Header title={"Войти в систему"} />
```
не верный синтаксис в пропсах

baklanovairina commented on src/store/store.js:
```diff
@@ -0,0 +1,86 @@
+  };
+
+  publish = (data) => {
+    const events = this.events;
```
отступ

Commit:
fix(common): :bug: fix classnames, code style, indents

Commit:
fix: correct interceptor

baklanovairina commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,16 @@
+import headerIcon from "src/images/headerIcon.svg";
+import "./style.scss";
+
+const Header = (props) => {
```
делай деструктуризацию

baklanovairina commented on src/components/Home/Home.jsx:
```diff
@@ -0,0 +1,39 @@
+        <button 
+          onClick={logOutUser} 
+          type="button" 
+          className="header__button">
```
код стайл

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,120 @@
+        <Header title="Войти в систему"/>
+      </div>
+      <div className="login-body">
+          <img 
```
табы поправь

baklanovairina commented on src/components/RegistrationForm/RegistrationForm.jsx:
```diff
@@ -0,0 +1,143 @@
+          <img 
+            src={bodyImage} 
+            alt="" 
+            className="registration-body-form__image" />
```
```
  className="registration-body-form__image" 
/>
```

baklanovairina commented on src/http/index.js:
```diff
@@ -0,0 +1,39 @@
+    const originalRequest = error.config;
+    if (
+      error.response.status === 401 &&
+      error.config &&
```
уже сделал переприсваивание, почему тогда error.config?

baklanovairina commented on src/http/index.js:
```diff
@@ -0,0 +1,39 @@
+        console.log("Не авторизирован");
+      }
+    }
+    throw error;
```
сплошной код, делай отступы, где нужно

baklanovairina commented on src/http/index.js:
```diff
@@ -0,0 +1,39 @@
+        localStorage.setItem("token", response.date.accessToken);
+        return $api.request(originalRequest);
+      } catch (error) {
+        console.log("Не авторизирован");
```
никаких консолей на фронте!

Commit:
fix: fix code style

Commit:
fix: fix error catch

baklanovairina commented on src/components/LogInForm/LogInForm.jsx:
```diff
@@ -0,0 +1,120 @@
+        <img 
+          src={bodyImage} 
+          alt="" 
+          className="login-body__image" />
```
```
  className="login-body__image" 
/>
```

Commit:
fix: fix code style
